src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java

```diff
@@ -269,6 +277,216 @@ private Optional<Pocket> findPocketForReload(ReloadBalanceRequest reloadRequest,
+        String operationUuid = revertRequest.getUuid();
+
+        Optional<BalanceChangeEventDto> duplicateOperation = findExistBalanceChangeEventDto(operationUuid, false);
+        if (duplicateOperation.isPresent()) {
+            return duplicateOperation.get();
+        }
+
+        log.info("Start revert balance operation with UUID: {}", revertRequest.getUuidToRevert());
+
+        BalanceChangeEventDto operationToRevert = findExistBalanceChangeEventDto(revertRequest.getUuidToRevert(), true)
+            .orElseThrow(() -> {
+                log.error("Not found balance change event with UUID: {}", revertRequest.getUuidToRevert());
+                return new EntityNotFoundException("Balance change event with UUID " + revertRequest.getUuidToRevert() + "not found");
+            });
+
+        boolean isRevertOperationAlreadyProvided = balanceChangeEventRepository.existsByRevertOperationId(revertRequest.getUuidToRevert());
+        if (isRevertOperationAlreadyProvided) {
+            log.error("Revert operation for UUID: {} - already provided", revertRequest.getUuidToRevert());
+            throw new BusinessException("error.revert.operation.already.provided", "Revert operation already provided");
+        }
+
+        OperationType previousOperationType = operationToRevert.getOperationType();
+        log.info("Operation type to revert: {}", previousOperationType);
+        if (previousOperationType.equals(RELOAD)) {
+            return revertReloadOperation(balance, revertRequest, operationToRevert);
+        } else if (previousOperationType.equals(CHARGING)) {
+            return revertChargingOperation(balance, revertRequest, operationToRevert);
+        } else {
+            log.error("Could not operation provide, because revert operation type: {} - not supported", operationToRevert.getOperationType());
+            throw new BusinessException("error.revert.operation.type.unknown", "Revert operation type not supported");
+        }
+    }
+
+    private BalanceChangeEventDto revertReloadOperation(Balance balance,
+                                                        RevertBalanceOperationRequest revertRequest,
+                                                        BalanceChangeEventDto reloadOperationToRevert) {
+        String operationUuid = StringUtils.isNotBlank(revertRequest.getUuid())
+            ? revertRequest.getUuid()
+            : UUID.randomUUID().toString();
+
+        BalanceChangeEvent changeEvent = createBalanceChangeEvent(
+            operationUuid,
+            reloadOperationToRevert.getAmountTotal(),
+            Metadata.of(revertRequest.getMetadata()),
+            now(clock),
+            balance,
+            true,
+            REVERT_RELOAD
+        );
+        changeEvent.setRevertOperationId(revertRequest.getUuidToRevert());
+
+        log.info("Revert reload mode: {}", revertRequest.getRevertReloadMode());
+        if (revertRequest.getRevertReloadMode() == null) {
```

Comment:
can we make `EXCEPTION` -> `DEFAULT` and make it the default mode?

Reply:
Fixed